darwin-bootstrap/posixspawn-hook.c

```diff
@@ -241,9 +255,9 @@ static int hook_posix_spawn_generic(__typeof__(posix_spawn) *old,
         /* append if it isn't one of ours */
         bool is_substitute =
             (next - p == sizeof(bl_dylib) - 1 &&
-             !memcmp(next, bl_dylib, sizeof(bl_dylib) - 1)) ||
+             !memcmp(p, bl_dylib, sizeof(bl_dylib) - 1)) ||
             (next - p == sizeof(psh_dylib) - 1 &&
-             !memcmp(next, psh_dylib, sizeof(psh_dylib) - 1));
+             !memcmp(p, psh_dylib, sizeof(psh_dylib) - 1));
```

Comment:
next is a pointer to the end, wherever we want to append the information, p[0..(next - p)] is the string that should be compared